### PR TITLE
In progress, making final adjustments to styling in css and giving de…

### PR DESCRIPTION
**Pull request by SharjeelSIqbal**
…tail to areas that have been overlooked.

### CODE REVIEW

**Comment by SharjeelSIqbal** [comment]
Kapture 2021-07-13 at 16 03 33
image
Kapture 2021-07-13 at 16 08 06

**Comment by SharjeelSIqbal** [comment]
Kapture 2021-07-13 at 16 10 29

File: js/main.js [hunk 1]
```diff
@@ -1,2 +1,44 @@
...
+
+$urlInput.addEventListener('blur', setImage);
+
+function resetForm(event) {
+  data.entries.push({
+    title: $title.value,
+    photoURL: $urlInput.value,
+    notes: $notes.value
+  });
+  data.nextEntryId += 1;
```
Comment (codingcodymiller) [inline js/main.js]: You did not add a `entryId` property to the new entry before incrementing this, please do so

File: js/main.js [hunk 2]
```diff
@@ -1,2 +1,44 @@
...
+
+  $title.value = '';
+  $urlInput.value = '';
+  $notes.value = '';
```
Comment (codingcodymiller) [inline js/main.js]: Instead of doing this, you should do `$submit.reset()`

File: js/main.js [hunk 3]
```diff
@@ -1,2 +1,44 @@
...
+  document.querySelector('img').setAttribute('src', 'images/placeholder-image-square.jpg');
+  event.preventDefault();
```
Comment (codingcodymiller) [inline js/main.js]: I generally prefer that `event.preventDefault()` is the first line of any function it needs to be a part of, for the sake of readability

File: js/main.js [hunk 4]
```diff
@@ -1,2 +1,44 @@
...
+
+}
+
+$submit.addEventListener('submit', resetForm);
+function getPreviousEntries(event) {
+  return localStorage.setItem('note-pages', JSON.stringify(data));
+}
```
Comment (codingcodymiller) [inline js/main.js]: This should only be done before the page unloads, not after every time the form is submitted. Also, all `localStorage` code should be in `data.js`, not `main.js`

File: js/main.js [hunk 5]
```diff
@@ -1,2 +1,44 @@
...
+
+var previousEntries = localStorage.getItem('note-pages');
+if (previousEntries !== null) {
+  previousEntries = JSON.parse(previousEntries);
+  data.entries = previousEntries.entries;
+  if (previousEntries.entries.length !== 1) {
+    data.nextEntryId = previousEntries.entries.length;
+  }
```
Comment (codingcodymiller) [inline js/main.js]: You should just resave the entire parsed object as the new value for `data`

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by SharjeelSIqbal** [comment]
Screen Shot 2021-07-13 at 5 11 36 PM

File: js/main.js [hunk 6]
```diff
@@ -13,32 +13,17 @@ function setImage(event) {
 $urlInput.addEventListener('blur', setImage);
 
 function resetForm(event) {
+  event.preventDefault();
   data.entries.push({
     title: $title.value,
     photoURL: $urlInput.value,
-    notes: $notes.value
+    notes: $notes.value,
+    entryId: data.entries.length
```
Comment (codingcodymiller) [inline js/main.js:21]: this should be `entryId: data.nextEntryId`

File: js/main.js [hunk 7]
```diff
@@ -13,32 +13,17 @@ function setImage(event) {
...
   });
-  data.nextEntryId += 1;
-
-  $title.value = '';
-  $urlInput.value = '';
-  $notes.value = '';
+  data.nextEntryId = data.entries.length;
```
Comment (codingcodymiller) [inline js/main.js:23]: This is wrong, you should just be incrementing the value by 1

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**